src/main/java/ch/njol/skript/effects/EffBan.java

```diff
@@ -73,6 +73,7 @@
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
 		players = exprs[0];
 		reason = exprs.length > 1 ? (Expression<String>) exprs[1] : null;
+		duration = exprs.length > 2 ? (Expression<String>) exprs[2] : null;

```

Comment:
Since this is a Timespan, it should be ``Expression<Timespan>`` and not ``Expression<String>``

Reply:
You are also missing the declaration of the ``duration`` variable, please append this to your code:
```
@Nullable
private Expression<Timespan> duration;
```

Reply:
Remember to import the Timestamp class also, via ```import ch.njol.skript.util.Timespan;```